extensions/amp-analytics/0.1/session-manager.js

```diff
@@ -77,18 +79,30 @@ export class SessionManager {
    * @return {!Promise<number|undefined>}
    */
   getSessionValue(type, value) {
-    return this.get(type).then((session) => {
-      return session?.[value];
+    return this.get(type).then((session) => session?.[value]);
+  }
+
+  /**
+   * Update EventTimestamp for this session async.
+   * Passes in a callback to set the EventTimestamp,
+   * to be used when the session it retrieved or created.
+   * @param {string} type
+   * @return {!Promise}
+   */
+  updateEventTimestamp(type) {
+    return this.get(type, (session) => {
```

Comment:
@alanorozco We don't need to return here, but I found that this is the only way to have the unit tests work (awaiting this method, `await macroTask()` always times-out for some reason). Do you think this is ok?

Reply:
Sure.